example/conf/sweeper/wandb.yaml

```diff
@@ -0,0 +1,69 @@
+# @package _global_
+defaults:
+  - override /hydra/sweeper: wandb
+
+hydra:
+  sweeper:
+    wandb_sweep_config:
+      name: hydra_plugin_test
+      metric:
+        name: out
+        goal: minimize
+      method: bayes
+      # number of agents to launch in a batch until budget is reached
+      num_agents: 2
+      budget: 4  # total number of agents to launch (must be >= num_agents)
+      # entity: your_entity
+      project: hydra_plugin_test_project
+      notes: ${hydra.overrides.task}
+      tags:
+        - ${experiment.name}
+        - ${model.name}
+        - ${experiment.db}
+      early_terminate:
+        type: hyperband
+        min_iter: 3
+        # max_iter: 27  #  the maximum number of iterations.
+        # s: 2  # the total number of brackets (required for max_iter)
+        eta: 3  # the bracket multiplier schedule
+
+      # Recommended to set to 1 when running in SLURM environment.
+      # Each agent only runs one training job then exits.
+      count: 3  # number of function evaluations to perform per agent
+
+      # maximum authorized failure rate for a batch of wandb agents' runs
+      max_run_failure_rate: 0.3
+
+      # maximum authorized failure rate for a batch of wandb agents launched
+      # by the launcher
+      max_agent_failure_rate: 0.1
+
+      # sweep_id:  # used for resuming from a previous sweep
+
+    # parameters are described using dot notation but will be accessible within
+    # the task function in typical OmegaConf fashion,
+    # e.g., model_param1 = cfg.model.param1
+    params:
+      # categorical, one or the other
+      experiment.db:
+        - mnist
+        - cifar
+
+      # a log-distributed positive scalar, evolving by factors of 2 on average
+      experiment.lr:
+        distribution: q_log_uniform
+        min: 0.02
+        max: 1
+        q: 2.0
+
+      # a linearly-distributed scalar between 0 and 1
+      model.dropout:
+        distribution: uniform
+        min: 0.0
+        max: 1.0
+
+      # a linearly-distributed integer scalar between 4 to 16
+      experiment.batch_size:
+        distribution: int_uniform
+        min: 4
+        max: 16
```

Comment:
That's what Hydra's meant for! All the other sweepers follow something similar. Furthermore, they're standardizing this approach further in v.1.2 https://github.com/facebookresearch/hydra/issues/1890

Reply:
Just a heads up that grid-type hyperparam sweeps still needs to be supported, hence the PR still being in draft state. I don't feel comfortable putting it in review state until I've got that sorted out since that's still an essential part of the wandb hyperparam sweep mechanism, lest we want an incomplete sweeper :)